content/source/docs/cloud/workspaces/variables.html.md

```diff
@@ -1,130 +1,106 @@
-
-You can also use the optional [Terraform Cloud Provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/variable) to update a workspace's variables. This has the same effect as managing workspace variables manually or via the API, but can be more convenient for large numbers of complex variables.
+[permissions-citation]: #intentionally-unused---keep-for-maintainers
 
-## Variable Limits
 
-The following limits apply to variables:
 
-Component   |  Limit
-------------|---------------------------
-description |  512 characters
-key         |  128 characters
-value       |  256 kilobytes
+## Types
 
-## Managing Variables in the UI
+You can create both Terraform variables and environment variables in Terraform Cloud.
 
-To view and manage a workspace's variables, navigate to that workspace and click the "Variables" navigation link at the top.
+> **Hands-on:** The [Create Infrastructure tutorial](https://learn.hashicorp.com/tutorials/terraform/cloud-workspace-configure?in=terraform/cloud-get-started) on HashiCorp Learn shows how to set both Terraform variables and provider credential environment variables.
 
-The variables page has separate lists of Terraform variables and environment variables:
+### Terraform Variables
 
-![Screenshot: The initial appearance of a workspace's variables page](./images/vars.png)
+Terraform variables refer to [input variables](/docs/language/values/variables.html) that define parameters without hardcoding them into the configuration. For example, you could create variables that let users specify the number and type of Amazon Web Services EC2 instances they want to provision with a Terraform module. If a required input variable is missing, Terraform plans in the workspace will fail and print an explanation in the log.
 
-To edit a variable, click one of its text fields or its pencil (edit) icon to reveal the editing controls. Make any desired changes to the variable's name, value, description, and settings, then click the "Save Variable" button.
+Terraform Cloud passes input variables to Terraform by writing a `terraform.tfvars` file, meaning that Terraform Cloud overwrites any `terraform.tfvars` file you check into version control. You should not check in `terraform.tfvars` files even when running Terraform solely on the command line.
 
-To add a variable, click the "+ Add Variable" button, enter a name, value, optional description, and save.
+You must define input variables manually; Terraform Cloud does not automatically discover variable names from Terraform code. However, workspaces that use Terraform 0.10.0 or later allow you to commit any number of [`*.auto.tfvars` files](/docs/language/values/variables.html#variable-files) to provide default Terraform input variables. If any variables from these files have the same key as existing variables in the Terraform Cloud workspace, the variables applied to the workspace overwrite variables from the files.
```

Comment:
"Terraform Cloud does not automatically discover variable names from Terraform code" is this true? I thought TFC would use the default value when you declare the variables in your config

Reply:
this is confusingly written - let me work this out and push up a fix. I need to confirm with eng :-)

Reply:
Okay I'm pushing up a big fix to this that I think makes it a lot clearer. Thank you for calling this out - this happened as I was reworking the page (pulling content from various places) and it didn't mesh quite right. I'm hoping what I'm pushing up is better!